src/uu/cut/src/cut.rs

```diff
@@ -337,6 +353,101 @@ fn cut_fields<R: Read>(reader: R, ranges: &[Range], opts: &FieldOptions) -> URes
     Ok(())
 }
 
+#[allow(clippy::cognitive_complexity)]
```

Comment:
I think clippy is actually making a good point here. I think this function should just delegate to the `cut_fields_whitespace`, `cut_fields_delimiter` and a new function. Ideally some of that could would also be deduplicated but we can worry about that later.

Reply:
I agree. I can do that in the next PR -- don't want to make this PR too convoluted :)

Reply:
Okay, but I don't see the warning from clippy if I run it locally. Did you get the warning or did you copy-paste the allow? And if you did, what version of clippy were you using?

Sidenote: please don't resolve conversations if you're not making changes. You're welcome to disagree of course, but then I'd like to resolve it once I think it's settled.

Reply:
I didn't add this line -- it was from the original code. See original code line 259.

Reply:
> Sidenote: please don't resolve conversations if you're not making changes. You're welcome to disagree of course, but then I'd like to resolve it once I think it's settled.

Oops sorry about that. Understood.

Reply:
I removed clippy allow, since it does not seem to trigger any warning

Reply:
Excellent! Thank you!